requirements.txt

```diff
@@ -2,6 +2,7 @@ anyconfig>=0.9.7, <1.0
 click<8.0
 cookiecutter>=1.6.0, <2.0
 memory_profiler>=0.50.0, <1.0
+networkx==2.3
```

Comment:
If there is no reason to hardpin a specific version, we would pin the upperbound to be the next major one.

Reply:
I set requirement to `networkx<2.4`. Would it be appropriate to add a lower bound as well? Or does `pip` grab the latest version?

Reply:
The common approach (as you can see above) would be something like `networkx>=2.3, <3.0`
The minor version changes are usually meant to be backward compatible, only add more features and fix existing bugs. The major version changes are often breaking, automatically installing them is likely to cause trouble.